### PR TITLE
Move bias addition before dropout

### LINKED ISSUE #187: Dropout between convolution and bias_add
Is there any reason to put dropout between convolution and bias_add? 
Generally speaking, we should do dropout after adding bias, because bias_add is part of convolution, right?

/////////////////////////////////////////////////////////
        conv1 = conv2d(in_node, w1, keep_prob)  
        tmp_h_conv = tf.nn.relu(conv1 + b1)

### PR DESCRIPTION
**Pull request by wkeithvan**
Fixes #187

### CODE REVIEW

**Comment by yiqianglee** [comment]
And for dropout issue I mentioned in #187 , you can add some debug code in below code, Google's design essential is to skip "dropout" when keep_prob is "1", but in our U-NET model, we use the "placeholder" to store the keep_prob, that will make "dropout" to execute every time, even you set keep_prob as 1, that's a big computation waster. 
The reason here is, "placeholder" is just a placeholder, during Tensorflow graph generation phase, we should not get the correct value, so this condition will always fail (keep_prob is none), my suggestion is to align google's design essential, not use "placeholder" here, what do you think?

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/nn_ops.py#L2316

**Comment by yiqianglee** [comment]
Sorry.. i have another question about dropout, in U-NET, we put dropout between conv and activation(relu), is there any special reason for that? Because when i see many other popular models, dropout should always be appended after activation (relu), like below mnist, is there any reason for you to put dropout between them? Training accuracy? Thanks!

https://github.com/tensorflow/models/blob/master/official/mnist/mnist.py#L84

**Comment by wkeithvan** [comment]
I am not actually in any way affiliated with this repository, I've just trying to use it for a segmentation problem and have been adding pull requests to fix bugs as I find them (or someone else like you finds them). I'm fairly new to ML and TensorFlow, so I have found actively contributing has helped me learn a lot fairly quickly and get used to getting my hands dirty in the nitty gritty of the code.

As for the dropout, I think you're right, but that is an architecture choice that might be better for @jakeret  to answer. Additionally, I noticed that the dropout is only ever applied on the downwards convolutions and never the upwards convolutions. Is this a design choice or should dropout also be added to `deconv2d ` function (or after its activation as @yiqianglee is suggesting)?

**Comment by jakeret** [comment]
Great, thank a lot for contributing to the package. Happy to merge the change into the code base.

As for the layer order: there is no well founded reason for the current design. It was just what I implemented back then. If one of you guys might have time to investigate and report if it has an effect (and create a PR) would be fantastic

**Comment by yiqianglee** [comment]
@jakeret @wkeithvan Thanks for your kindly reply, i will try to make a PR for review.
